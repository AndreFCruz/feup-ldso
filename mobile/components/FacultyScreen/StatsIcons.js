import React from 'react';
import {View} from 'react-native';
import {Text, Icon} from 'native-base';
import PropTypes from 'prop-types';
import {widthPercentageToDP as wp, heightPercentageToDP as hp} from 'react-native-responsive-screen';

class StatisticsNumbers extends React.Component {

    constructor(props) {
        super(props);
    }

    render() {
        const {icon, iconsNmb,  percentage, text} = this.props;
        
        var icons = [];
        let colored = 0;

        for(let i = 0; i < iconsNmb; i++) {

            if(colored < (iconsNmb*percentage/100)) {
                icons.push(
                    <Icon key={i} style={styles.icon} type="FontAwesome" name={icon}/>
                );
                colored++;
            } else {
                icons.push(
                    <Icon key={i} style={styles.backgroundIcon} type="FontAwesome" name={icon}/>
                );
            }
        }    

        return (
            <View style={styles.main}>
                <View style={[styles.section, styles.justify]}>
                    {icons}
                </View>
                <View style={[styles.section, styles.percentageContent]}>
                    <Text style={styles.percentage}>{percentage}%</Text>
                </View>
                <View style={styles.section}>
                    <Text style={styles.description}>{text}</Text>
                </View>
            </View>
        );
    }
}

StatisticsNumbers.propTypes = {
    icon: PropTypes.string,
    iconsNmb: PropTypes.number,
    percentage: PropTypes.number,
    text: PropTypes.string
};

export default StatisticsNumbers ;

const styles = {
    main: {
        flex: 1,
        flexDirection: 'column',
        padding: wp('1%'),
        paddingBottom: hp('0.5%'),
        alignItems: 'center',
        justifyContent: 'center',
    },

    section: {
        flex:1,
        flexDirection: 'row',
        alignItems: 'center',
        justifyContent: 'center',
        flexWrap: 'wrap',
    },

    justify: {
        justifyContent: 'center'
    },

    backgroundIcon: {
        fontSize: hp('9%'),
        padding: hp('0.5%'),
        color: '#cccccc'
    },

    icon: {
        fontSize: hp('9%'),
        padding: hp('0.5%'),
        color: '#1c1c1c'
    },

    percentage: {
        fontFamily: 'Quicksand_regular',
        fontSize: hp('5%'),
        color: 'black',
    },

    percentageContent: {
        paddingTop: hp('23.5%'),
        marginBottom: hp('5%'),
    },

    
    description: {
        justifyContent: 'center',
        alignItems: 'center',
        fontFamily: 'Quicksand_regular',
        fontSize: wp('1.6%'),
        textAlign: 'center'

    }

};