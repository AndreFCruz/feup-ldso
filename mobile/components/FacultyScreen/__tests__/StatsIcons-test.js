import React from 'react';
import StatsIcons  from '../StatsIcons';
import {Icon} from 'native-base';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

function setup() {
    const wrapper = shallow(<StatsIcons icon='flask' iconsNmb={12} percentage={50} text="Teachers and Researchers"/>,
        {
            disableLifecycleMethods: true
        });
    return wrapper;
}

describe('Rendering', () => {
    it('Render component', () => {
        const wrapper = setup();
        expect(wrapper.dive()).toMatchSnapshot();
    });
    it('check number of icons', () =>{
        const wrapper = setup();
        expect(wrapper.find(Icon)).toHaveLength(12);
    });
    it('check black icons', () => {
        const wrapper = setup();
        expect(wrapper.find('.black')).toHaveLength(6);
    });
});
