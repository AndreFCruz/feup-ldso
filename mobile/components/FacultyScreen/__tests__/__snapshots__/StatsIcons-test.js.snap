// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering Render component 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "center",
      "padding": 7.5,
      "paddingBottom": 6.5,
    }
  }
>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "center",
        },
        Object {
          "justifyContent": "center",
        },
      ]
    }
  >
    <Styled(Icon)
      key="0"
      name="flask"
      style={
        Object {
          "color": "#1c1c1c",
          "fontSize": 37.5,
          "padding": 1,
        }
      }
      type="FontAwesome"
    />
    <Styled(Icon)
      key="1"
      name="flask"
      style={
        Object {
          "color": "#1c1c1c",
          "fontSize": 37.5,
          "padding": 1,
        }
      }
      type="FontAwesome"
    />
    <Styled(Icon)
      key="2"
      name="flask"
      style={
        Object {
          "color": "#1c1c1c",
          "fontSize": 37.5,
          "padding": 1,
        }
      }
      type="FontAwesome"
    />
    <Styled(Icon)
      key="3"
      name="flask"
      style={
        Object {
          "color": "#1c1c1c",
          "fontSize": 37.5,
          "padding": 1,
        }
      }
      type="FontAwesome"
    />
    <Styled(Icon)
      key="4"
      name="flask"
      style={
        Object {
          "color": "#1c1c1c",
          "fontSize": 37.5,
          "padding": 1,
        }
      }
      type="FontAwesome"
    />
    <Styled(Icon)
      key="5"
      name="flask"
      style={
        Object {
          "color": "#1c1c1c",
          "fontSize": 37.5,
          "padding": 1,
        }
      }
      type="FontAwesome"
    />
    <Styled(Icon)
      className="black"
      key="6"
      name="flask"
      style={
        Object {
          "color": "#cccccc",
          "fontSize": 37.5,
          "padding": 1,
        }
      }
      type="FontAwesome"
    />
    <Styled(Icon)
      className="black"
      key="7"
      name="flask"
      style={
        Object {
          "color": "#cccccc",
          "fontSize": 37.5,
          "padding": 1,
        }
      }
      type="FontAwesome"
    />
    <Styled(Icon)
      className="black"
      key="8"
      name="flask"
      style={
        Object {
          "color": "#cccccc",
          "fontSize": 37.5,
          "padding": 1,
        }
      }
      type="FontAwesome"
    />
    <Styled(Icon)
      className="black"
      key="9"
      name="flask"
      style={
        Object {
          "color": "#cccccc",
          "fontSize": 37.5,
          "padding": 1,
        }
      }
      type="FontAwesome"
    />
    <Styled(Icon)
      className="black"
      key="10"
      name="flask"
      style={
        Object {
          "color": "#cccccc",
          "fontSize": 37.5,
          "padding": 1,
        }
      }
      type="FontAwesome"
    />
    <Styled(Icon)
      className="black"
      key="11"
      name="flask"
      style={
        Object {
          "color": "#cccccc",
          "fontSize": 37.5,
          "padding": 1,
        }
      }
      type="FontAwesome"
    />
  </Component>
  <Component
    style={
      Array [
        Object {
          "alignItems": "center",
          "flex": 1,
          "flexDirection": "row",
          "flexWrap": "wrap",
          "justifyContent": "center",
        },
        Object {
          "flex": 1,
          "flexDirection": "row",
          "marginBottom": 40,
          "paddingTop": 313.5,
        },
      ]
    }
  >
    <Styled(Text)
      style={
        Object {
          "color": "black",
          "flexDirection": "row",
          "fontFamily": "Quicksand_regular",
          "fontSize": 66.5,
        }
      }
    >
      50
      %
    </Styled(Text)>
  </Component>
  <Component
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "flexDirection": "row",
        "flexWrap": "wrap",
        "justifyContent": "center",
      }
    }
  >
    <Styled(Text)
      style={
        Object {
          "alignItems": "center",
          "fontFamily": "Quicksand_regular",
          "fontSize": 12,
          "justifyContent": "center",
          "textAlign": "center",
        }
      }
    >
      Teachers and Researchers
    </Styled(Text)>
  </Component>
</View>
`;
